ouroboros-network/src/Ouroboros/Network/BlockFetch/State.hs

```diff
@@ -118,20 +127,25 @@ fetchLogicIteration
   -> FetchTriggerVariables peer header m
   -> FetchNonTriggerVariables peer header block m
   -> FetchStateFingerprint peer header block
+  -> ConsensusFetchStateRefinementPreBlockFetch peer header block m
   -> m (FetchStateFingerprint peer header block)
 fetchLogicIteration decisionTracer clientStateTracer
                     fetchDecisionPolicy
                     fetchTriggerVariables
                     fetchNonTriggerVariables
-                    stateFingerprint = do
+                    stateFingerprint
+                    consensusRefinement = do
 
     -- Gather a snapshot of all the state we need.
     (stateSnapshot, stateFingerprint') <-
-      atomically $
-        readStateVariables
+      atomically $ do
+        (state, fingerprint) <- readStateVariables
           fetchTriggerVariables
           fetchNonTriggerVariables
           stateFingerprint
+        state' <- consensusRefinement state
```

Comment:
If we can't find a more descriptive name than `consensusRefinement` it might be worthwhile commenting on the relation between `state` and `state'`.